cloud/amazon/ec2_group.py

```diff
@@ -163,6 +163,10 @@ def validate_rule(module, rule):
     VALID_PARAMS = ('cidr_ip',
                     'group_id', 'group_name', 'group_desc',
                     'proto', 'from_port', 'to_port')
+
+    if not isinstance(rule, dict):
```

Comment:
Since this module requires python-2.6 or higher, we should use the ABCs to check whether something is "dict-like".
``` python
from collections import Mapping
[...]
    if not isinstance(rule, Mapping):
```

note that there's also a MutableMapping ABC if this interface needs to write to the dict as well as read from it.